Update of the doc of simpl and cbn, especially wrt "simpl never". (#18602)
coq/coq: doc/sphinx/proofs/writing-proofs/equality.rst

.. tacn:: cbn {? @reductions } @simple_occurrences

>    :tacn:`cbn` is a variant of :tacn:`cbn` implemented as an abstract machine.

Reviewer: Not sure this is much more useful to the manual reader than the previous comment.
Typically, the user still don't know when and why she should prefer `cbn` over `simpl` (or vice versa).

Author: Yes! And to be honest, I still don't know myself. Hopefully, the ongoing discussions will help.

Reviewer 2: Shouldn't you be comparing simpl and cbn here?  I don't see how "cbn is a variant of cbn" makes any sense.

Reviewer: @herbelin apparently, the discussion is not happening / leading to any conclusion, why not reverting that sentence and merge the remaining of the PR?

Reviewer: Let's merge as it is without this sentence since the discussion is not progressing on it.

   The main difference is that :tacn:`cbn` may unfold constants even when they
   cannot be reused in recursive calls: in the previous example, :g:`succ t` is
   reduced to :g:`S t`. Modifiers such as `simpl never` are also not treated the same,
   see :ref:`Args_effect_on_unfolding`.

   Setting :opt:`Debug` ``"RAKAM"`` makes :tacn:`cbn` print various debugging information.
   ``RAKAM`` is the Refolding Algebraic Krivine Abstract Machine.

   .. example::

>       Here are typical examples showing a difference or absence of
>       difference between :tacn:`cbn` and :tacn:`simpl`:

Reviewer 2: ```suggestion
      Here are typical examples comparing :tacn:`cbn` and :tacn:`simpl`:
```
